nearcore: 1.25.0 -> 1.26.0

Mic92 opened the pull request:
###### Description of changes

Release is expected to come out today.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Mic92 committed:
nearcore: 1.25.0 -> 1.26.0

Mic92 commented:
cc @mikroskeem

Mic92 commented:
@ofborg build nearcore

mikroskeem reviewed (approved):
LGTM

mikroskeem commented on pkgs/applications/blockchains/nearcore/default.nix:
```diff
@@ -1,32 +1,46 @@
+
+  postPatch = ''
+    substituteInPlace neard/src/main.rs \
+      --replace 'git_version!(fallback = "unknown")' '"nix:${version}"' \
```
One side note though: I'm not sure if you want to do this in official nixpkgs, might be better idea to include .git directory and let build script do the rest?

Mic92 replied:
Having `.git` introduces a lot of headaches because it makes the whole thing non-reproducible and leads to breaking checksums when fetching a repo.

mikroskeem commented on pkgs/applications/blockchains/nearcore/default.nix:
```diff
@@ -1,32 +1,46 @@
+  '';
+
+  CARGO_PROFILE_RELEASE_CODEGEN_UNITS = "1";
+  CARGO_PROFILE_RELEASE_LTO = "thin";
```
One more thing - nearcore Makefile is using [fat LTO](https://github.com/near/nearcore/blob/c164b7df5f94f54c8f36305348dcd9f9b9994ddc/Makefile#L2=) officially. Might want to match that as well?

Mic92 replied:
How much memory does require? I don't want reviewers machine / builds to go out of memory.

Mic92 replied:
~3644M on my machine.

Mic92 replied:
I enabled it. Does not seem to heavy.

Mic92 committed:
nearcore: mark as linux only for now

macos support is feasible but requires some testing: https://github.com/ZentriaMC/neard-nix/blob/master/neard.nix#L23

Mic92 commented:
I disabled macos support for now. When can enable it if we have a tester for this platform.

Mic92 commented:
@ofborg build nearcore

Mic92 committed:
nearcore: switch to fat lto

Mic92 committed:
nearcore: add mikroskeem as a maintainer

Mic92 commented:
@GrahamcOfBorg build nearcore

Mic92 commented:
```
╭─[~/git/nixpkgs]─[nix-update]─[rebase-i]─[11m 31s]
╰─ % ./result/bin/neard --version
neard (release 1.26.0) (build nix:1.26.0) (rustc 1.60.0) (protocol 53) (db 31)
```
